spec/controllers/foods_spec.rb

```diff
@@ -0,0 +1,99 @@
+require 'rails_helper'
+
+RSpec.describe FoodsController, type: :controller do
+  describe 'GET #index' do
+    context 'when user is authenticated' do
+      before do
+        @user = User.create(name: 'Test user', email: 'test444@gmail.com',
+                            password: '123456', password_confirmation: '123456')
+        sign_in @user
+        get :index
+      end
+
+      it 'assigns @foods with user foods' do
+        expect(assigns(:foods)).to eq(@user.foods)
+      end
+
+      it 'renders the index template' do
+        expect(response).to render_template(:index)
+      end
+
+      it 'returns a successful response' do
+        expect(response).to be_successful
+      end
+    end
+
+    context 'when user is not authenticated' do
+      before { get :index }
+
+      it 'redirects to the sign-in page' do
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'when user is authenticated' do
+      before do
+        @user = User.create(name: 'Test user', email: 'test444@gmail.com',
+                            password: '123456', password_confirmation: '123456')
+        sign_in @user
+      end
+
+      context 'with valid parameters' do
+        let(:valid_params) do
+          { food: { name: 'Test food', measurement_unit: 'pce', price: 12.2, quantity: 4 } }
+        end
+
+        it 'creates a new food' do
+          expect do
+            post :create, params: valid_params
+          end.to change(Food, :count).by(1)
+        end
+
+        it 'sets a flash notice' do
+          post :create, params: valid_params
+          expect(flash[:notice]).to eq('Food was successfully created.')
+        end
+      end
+
+      context 'with invalid parameters' do
+        # Add test cases for invalid parameters (if applicable)
+      end
+    end
+
+    context 'when user is not authenticated' do
+      it 'redirects to the sign-in page' do
+        post :create
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    context 'when user is authenticated' do
+      before do
+        @user = User.create(name: 'Test user', email: 'test444@gmail.com',
+                            password: '123456', password_confirmation: '123456')
+        sign_in @user
+        @food = Food.create(name: 'Test food', measurement_unit: 'pce', price: 12.2, quantity: 4, user: @user)
+      end
+
+      it 'destroys the food' do
+        expect do
+          delete :destroy, params: { id: @food.id }
+        end.to change(Food, :count).by(-1)
+      end
+
+      it 'redirects to the foods index' do
+        delete :destroy, params: { id: @food.id }
+        expect(response).to redirect_to(foods_path)
+      end
+
+      it 'sets a flash notice' do
+        delete :destroy, params: { id: @food.id }
+        expect(flash[:notice]).to eq('Food was successfully destroyed.')
+      end
+    end
+  end
```

Comment:
- [ ] 👍 Great job fixing almost all the changes I requested earlier, but I think you missed one required change. You tested your controllers and views, but you left out the models. I left the comment in the review comment area because there was nowhere in the code to show it.

![tests](https://github.com/ikoote1/Recipe-App/assets/53124548/579b6839-a656-4a35-b317-89bfc20904bb)